validation/src/main/java/io/serverlessworkflow/validation/SwitchStateValidator.java

```diff
@@ -0,0 +1,163 @@
+            error.setType(ValidationError.WORKFLOW_VALIDATION);
+            errors.add(error);
+        }
+
+        return errors;
+    }
+
+    private Optional<ValidationError> validateDefaultConditionNextStateNotFound(
+            Workflow workflow, String switchStateName, String nextState) {
+        if (nextState != null && !stateExists(workflow, nextState) && !"end".equals(nextState)) {
+            return Optional.of(error(String.format(
+                    NEXT_STATE_NOT_FOUND_FOR_DEFAULT_CONDITION_ERROR,
+                    nextState,
+                    switchStateName,
+                    workflow.getName())));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private static boolean stateExists(Workflow workflow, String state) {
+        List<State> states = workflow.getStates();
+        if (states != null) {
+            return states.stream()
+                           .map(State::getName)
+                           .filter(Objects::nonNull)
+                           .anyMatch(name -> name.equals(state));
+        } else {
+            return false;
+        }
+    }
+
+    private Optional<ValidationError> validateEventTimeoutDuration(
+            Workflow workflow, String timeout) {
+        if (isInvalidDuration(timeout)) {
+            return Optional.of(error(String.format(INVALID_EVENT_TIMEOUT_ERROR, timeout, workflow.getName())));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private static boolean isInvalidDuration(String timeout) {
+        try {
+            Duration.parse(timeout);
+            return false;
+        } catch (DateTimeParseException e) {
+            return true;
+        }
+    }
+
+    private static boolean switchStateHasEventCondition(SwitchState switchState) {
+        return switchState.getEventConditions() != null && !switchState.getEventConditions().isEmpty();
+    }
+
+    private static Optional<String> resolveEventTimeout(SwitchState switchState, Workflow workflow) {
+        return resolveStateTimeout(switchState).or(() -> resolveWorkflowTimeout(workflow));
+    }
+
+    private static Optional<String> resolveStateTimeout(SwitchState switchState) {
+        return Optional.ofNullable(switchState.getTimeouts()).map(TimeoutsDefinition::getEventTimeout);
+    }
+
+    private static Optional<String> resolveWorkflowTimeout(Workflow workflow) {
+        return Optional.ofNullable(workflow.getTimeouts()).map(TimeoutsDefinition::getEventTimeout);
+    }
+
+    private Optional<ValidationError> validateNextState(
+            Workflow workflow, String switchStateName, DefaultConditionDefinition defaultCondition) {
+        String nextState =
+                defaultCondition.getTransition() != null
+                        ? defaultCondition.getTransition().getNextState()
+                        : null;
+        if (nextState != null && !nextState.isEmpty()) {
```

Comment:
I think it will be nice to have an utility static method that check if a string is not null and is not empty, so you save a lot of code and dummy erros (in this case you forget the trim, in other case you miss the !)